native/packages/react-native-bpk-component-carousel/src/BpkCarousel.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Fragment, type Element, type ChildrenArray } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ScrollView,
+  type StyleObj,
+  ViewPropTypes,
+  StyleSheet,
+} from 'react-native';
+import BpkPaginationDots from 'react-native-bpk-component-pagination-dots';
+import { spacingXl } from 'bpk-tokens/tokens/base.react.native';
+import typeof BpkCarouselItem from './BpkCarouselItem';
+
+const styles = StyleSheet.create({
+  pagination: {
+    alignSelf: 'center',
+    marginTop: -spacingXl,
+  },
+});
+
+export type Props = {
+  accessibilityLabel: string | ((number, number) => string),
+  children: ChildrenArray<Element<BpkCarouselItem>>,
+  style: ?StyleObj,
+  renderPagination: ?boolean,
+};
+
+type State = {
+  currentIndex: number,
+  width: ?number,
+  height: ?number,
+};
+
+class BpkCarousel extends React.Component<Props, State> {
+  static propTypes = {
+    accessibilityLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+      .isRequired,
+    children: PropTypes.node.isRequired,
+    style: ViewPropTypes.style,
+    renderPagination: PropTypes.bool,
```

Comment:
@matthewdavidson is renaming `BpkPaginationDots` to `BpkCarouselIndicator` (I think, I may have gotten the name wrong), so once that happens, calling this prop `renderPagination` will be a bit confusing. How about `showIndicator` instead?

Reply:
yeah, I think that's probably better. will do.